### PR TITLE
Add sys info printout for all nodes

### PR DESCRIPTION
**Pull request by opotowsky**
## What is the change?

This adds a little extra info to the runLog for cases. In addition to the machine information, there's a `systeminfo` printout included for each node to show the processor type and OS info.

### For the reviewer:

I do not think this is an important enough update to include in the release notes, but will add it if the you feel otherwise. Additionally, I am not able to add a test for this unless I can add this to an MPI test file.

## Why is the change being made?

This helps prove that a case was run on a particular system. This is motivated by downstream analysis needs.

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [ ] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] No [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were altered.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/bookkeeping/report/reportingUtils.py [hunk 1]
```diff
@@ -188,6 +195,8 @@ def _writeMachineInformation():
                     tablefmt="armi",
                 )
             )
+            runLog.header("=========== System Information ===========")
+            runLog.info(sysInfo)
```
Comment (john-science) [inline armi/bookkeeping/report/reportingUtils.py:199]: Perhaps we can wrap these two lines in a check?

```python
if sysInfo:
    runLog.header("=========== System Information ===========")
    runLog.info(sysInfo)
``

File: armi/bookkeeping/report/reportingUtils.py [hunk 2]
```diff
@@ -180,6 +182,11 @@ def _writeMachineInformation():
                 nodeMappingData.append(
                     (uniqueName, numProcessors, ", ".join(matchingProcs))
                 )
+                # Run sys info on each unique node too
```
Comment (john-science) [inline armi/bookkeeping/report/reportingUtils.py:185]: Can we change this comment to something like:

```python
# If this is on Windows: run sys info on each unique node too
```

File: armi/bookkeeping/report/reportingUtils.py (hunk 1 above)
Comment (opotowsky) [reply armi/bookkeeping/report/reportingUtils.py:199]: done!

File: armi/bookkeeping/report/reportingUtils.py (hunk 2 above)
Comment (opotowsky) [reply armi/bookkeeping/report/reportingUtils.py:185]: done!

**Review by john-science — APPROVED** [review]

**Review by john-science — APPROVED** [review]

**Pull request merged**